source/framefmt.c

```diff
@@ -271,6 +300,7 @@ int aws_cryptosdk_serialize_frame(
         return aws_raise_error(result);
     } else {
         *ciphertext_buf = state.u.buffer;
+        AWS_POSTCONDITION(aws_cryptosdk_frame_is_valid(frame));
```

Comment:
If we're adding postconditions, I would suggest adding a postcondition that the ciphertext_buf is advanced by exactly `state.ciphertext_size` on success.

Reply:
Thanks for the suggestion! At the moment we don't have the infrastructure to reason about the initial (before the function call) value of a variable at the postconditions. However, I have added this assertion in the [harness](https://github.com/aws/aws-encryption-sdk-c/pull/378/files/e74d52d7c1595808171d79e6b6a6c2f029449724#diff-ac6b8365f754b91ff670c36d2435058dR44).